trinity/sync/common/chain.py

```diff
@@ -245,12 +262,18 @@ def register_peer(self, peer: BasePeer) -> None:
 
         return tail_headers
 
+    async def pop_all_pending_headers(self) -> Tuple[BlockHeader, ...]:
+        """Get all the currently pending headers. If no headers pending, wait until one is"""
+        queue = self.pending_headers
+        if queue.empty():
+            first_header = await queue.get()
+        else:
+            first_header = queue.get_nowait()
+
+        available = queue.qsize()
+        return (first_header, ) + tuple(queue.get_nowait() for _ in range(available))
```

Comment:
This function is clever enough that I think it deserves an initial comment.  It wasn't clear to me why it was written this way on my first read.

I.E. why get `first_header` and then pull the remaining tail headers...... Oh! so that we know there *was* something in the queue and still get the benefit of not doing a bunch of await calls.